src/boot/efi/boot.c

```diff
@@ -2707,9 +2728,9 @@ static EFI_STATUS run(EFI_HANDLE image) {
                                 "No loader found. Configuration files in \\loader\\entries\\*.conf are needed.");
 
         /* select entry or show menu when key is pressed or timeout is set */
-        if (config.force_menu || config.timeout_sec > 0)
+        if (config.force_menu || (config.timeout_sec > 0 && config.timeout_sec != TIMEOUT_MENU_DISABLED))
                 menu = true;
-        else {
+        else if (config.timeout_sec != TIMEOUT_MENU_DISABLED) {
```

Comment:
I think the correct approach is to reduce the timeout to 0 or 1 instead of completely preventing the menu.

Reply:
A timeout is passed to console_key_read. You can obviously reduce that instead of never even trying to get input.

Reply:
Disabled means disabled - on my headless devices, I do not want to lose any boot time for a menu I am not able to see nor use